pkg/generator/template.go

```diff
@@ -239,7 +248,16 @@ func (b *{{.Name|lcFirst}}Batch) startTimer(l *{{.Name}}) {
 }
 
 func (b *{{.Name|lcFirst}}Batch) end(l *{{.Name}}) {
+	defer func() {
+		if r := recover(); r != nil {
+			if l.recover != nil {
+				b.error = []error{l.recover(r)}
+			} else {
+				b.error = []error{fmt.Errorf("%v", r)}
+			}
+		}
+		close(b.done)
```

Comment:
this line will cause "panic: close of closed channel", should inside the if-recover case.

Reply:
Have you tried this code and run into it? The channel needs to always be closed when the function exits. Otherwise `Load` calls will never return.

We've since forked this code and are running panic-handling almost identical to this in production with no issues.

Reply:
Hi, i have tried the patch, but I got  "panic: close of closed channel".
From my opinion, it's because if l.fetch(b.keys) successfully execute, the channel will be closed, but inside defer func, we close it again. So, the second time we close it, we have to make sure there was panic, means inside r := recover(); && r != nil.